common/route.c

```diff
@@ -113,6 +109,8 @@ static bool dijkstra_to_hops(struct route_hop **hops,
 	if (!dijkstra_to_hops(hops, gossmap, dij, next, amount, cltv))
 		return false;
 
+	gossmap_chan_get_capacity(gossmap, c, &total);
+	(*hops)[num_hops].total_amount.millisatoshis = total.satoshis * 1000; /* Raw: simpler. */
```

Comment:
Please use amount_sat_to_msat().  I'm happy for you to abort or return false, but this just silently overflows.